src/dripper/polkadot/PolkadotActions.ts

```diff
@@ -1,223 +1,165 @@
+      client,
+      nonce: this.#nonce++,
+    });
+
+    logger.debug(`Teleporting to ${address}: ${parachain_id}. Transaction ${tx}`);
+
+    // client.submit(tx) waits for the finalized value, which might be important for real money,
+    // but for the faucet drips, early respond is better UX
+    const submit$ = client.submitAndWatch(tx);
+
+    const hash = (await firstValueFrom(submit$.pipe(filter((value) => value.type === "broadcasted")))).txHash;
+
+    await firstValueFrom(submit$.pipe(filter((value) => value.type === "txBestBlocksState" && value.found)));
+
+    logger.info(`💸 teleporting tokens to ${address}, parachain ${parachain_id}: done: ${hash}`);
+    return { hash };
   }
 
-  async sendTokens(address: string, parachain_id: string, amount: bigint): Promise<DripResponse> {
+  async transferTokens(dripAmount: bigint, address: string): Promise<DripResponse> {
+    logger.info(`💸 sending tokens to ${address}`);
+    const tx = await networkData.api.getTransferTokensTx({
+      dripAmount,
+      address,
+      client,
+      nonce: this.#nonce++,
+    });
+    logger.debug(`Dripping to ${address}. Transaction ${tx}`);
+    // client.submit(tx) waits for the finalized value, which might be important for real money,
+    // but for the faucet drips, early respond is better UX
+
+    const submit$ = client.submitAndWatch(tx);
+    submit$.forEach((data) => {
+      logger.info(`submit$ ${address}`, data.type);
+    });
+
+    const hash = (await firstValueFrom(submit$.pipe(filter((value) => value.type === "broadcasted")))).txHash;
+
+    await firstValueFrom(submit$.pipe(filter((value) => value.type === "txBestBlocksState" && value.found)));
+
+    logger.info(`💸 sending tokens to ${address}: done: ${hash}`);
+    return { hash };
+  }
+
+  async sendTokens(address: string, parachain_id: number | null, amount: bigint): Promise<DripResponse> {
     let dripTimeout: ReturnType<typeof rpcTimeout> | null = null;
     let result: DripResponse;
-    const faucetBalance = this.getFaucetBalance();
 
     try {
-      if (!this.account) throw new Error("account not ready");
-
-      if (typeof faucetBalance !== "undefined" && amount >= faucetBalance) {
+      if (typeof this.#faucetBalance !== "undefined" && amount >= this.#faucetBalance) {
         const formattedAmount = formatAmount(amount);
-        const formattedBalance = formatAmount(faucetBalance);
+        const formattedBalance = formatAmount(this.#faucetBalance);
 
         throw new Error(
-          `Can't send ${formattedAmount} ${networkData.currency}s, as balance is only ${formattedBalance} ${networkData.currency}s.`,
+          `Can't send ${formattedAmount} ${networkData.data.currency}s, as balance is only ${formattedBalance} ${networkData.data.currency}s.`,
         );
       }
 
       // start a counter and log a timeout error if we didn't get an answer in time
       dripTimeout = rpcTimeout("drip");
-      if (parachain_id != "") {
+      if (parachain_id !== null) {
```

Comment:
Wouldn't `!!parachain_id` obtain the same result?

Reply:
If I need to reduce `number | null` to `null`, I prefer to do it explicitly